tests/unit/test_filehandler.py

```diff
@@ -176,7 +181,12 @@ def test_timed_rotatingfilehandler_rollover(tmp_path, utc):
     handler.rollover_at = time.time() - 1
 
     logger.warning("test")
-    for file_name in os.listdir(tmp_path):
+    handler.close()
+
+    files = os.listdir(tmp_path)
+    assert len(files) == 2
+
+    for file_name in files:
         with open(tmp_path / file_name, "r") as file:
             assert file.read() == "test\n"
```

Comment:
Strangely enough I can't reproduce this but I remember the issue so I don't think that's related to timing.
Looks like the Rotation is happening and the new file is crated, so `log.txt` is renamed to `log.txt.12345` but the problem is the new file `log.txt.1234` is empty at the time we check it.

This is a wild guess but I'm thinking this is down to different OS behaviours, as I'm pretty sure it didn't fail on Ubuntu but failed in macOS and Windows.
So I think the rotation logic is working fine and we can assure that the files are created but the file is empty when we check it.

So maybe we can limit platforms for this test? Or avoid the `assert file.read() == "test\n"` check?